src/android/usb_host/android_uvc.cpp

```diff
@@ -0,0 +1,1804 @@
+                return h;
+            } else {
+                curr_index++;
+            }
+        }
+        h = usb_descriptor_iter_next(&iter);
+    }
+    return h;
+}
+
+uvc_error_t usbhost_open(usbhost_uvc_device *device, int InterfaceNumber) {
+    usbhost_uvc_interfaces *interfaces = NULL;
+    usb_config_descriptor *cfgDesc = NULL;
+    uint8_t *descriptors = NULL;
+    uvc_error_t ret = UVC_SUCCESS;
+
+    //LOG_DEBUG("usbhost_open() device: %s", device->deviceHandle->dev_name);
+    device->streams = NULL;
+
+    // Start by clearing deviceData, otherwise
+    // usbhost_uvc_free_device_info(&device->deviceData);
+    // will do something not good
+    memset(&device->deviceData, 0, sizeof(usbhost_uvc_device_info_t));
+
+    if (!device->device) {
+        LOG_ERROR("usb_device_new() failed to create usb device!");
+        ret = UVC_ERROR_INVALID_PARAM;
+        goto fail;
+    }
+
+
+    // Returns IVCAM configuration descriptor
+    cfgDesc = (usb_config_descriptor *) usbhost_get_descriptor(device->device->get_handle(),
+                                                               USB_DT_CONFIG,
+                                                               0);
+    if (cfgDesc == NULL) {
+        LOG_ERROR("usb_device_new() failed cannot get config descriptor!");
+        ret = UVC_ERROR_INVALID_PARAM;
+        goto fail;
+    }
+    device->deviceData.config = *cfgDesc;
+    //LOG_DEBUG("usbhost_open() parsing descriptors: %s", device->deviceHandle->dev_name);
+    // Iterate over all descriptors and parse all Interface and Endpoint descriptors
+    usb_descriptor_iter it;
+    usb_descriptor_iter_init(device->device->get_handle(), &it);
+    usbhost_uvc_parse_config_descriptors(&it,
+                                         &interfaces);
+    //LOG_DEBUG("usbhost_open() parsing descriptors complete found %d interfaces",interfaces->numInterfaces);
+
+    device->deviceData.interfaces = interfaces;
+
+
+    //LOG_DEBUG("usbhost_open() usbhost_uvc_scan_control");
+
+    // Fill fields of uvc_device_info on device
+    ret = usbhost_uvc_scan_control(device, &device->deviceData, InterfaceNumber);
+    if (ret != UVC_SUCCESS) {
+        LOG_ERROR("uvc_scan_control failed\n");
+        goto fail;
+    }
+
+    return ret;
+
+    fail:
+
+    if (descriptors) {
+        delete descriptors;
+    }
+
+    if (device) {
+        usbhost_uvc_free_device_info(&device->deviceData);
+
+        if (device->device) {
+            device->device = NULL;
+        }
+    }
+    LOG_ERROR("usbhost_open() failed! " << ret);
+    return ret;
+}
+
+
+uvc_error_t usbhost_close(usbhost_uvc_device *device) {
+    uvc_error_t ret = UVC_SUCCESS;
+
+    if (device != NULL) {
+        usbhost_uvc_free_device_info(&device->deviceData);
+        memset(&device->deviceData, 0, sizeof(usbhost_uvc_device_info_t));
+
+
+        if (device->device) {
+            //usb_device_close(device->deviceHandle->GetHandle()); //TODO: do we need to do this?
```

Comment:
Given that it closes down ok and it's been commented out, I guess the answer is no?

Reply:
Probably. Gone in 2bca5302b